httpmw/rbac_casbin.go

```diff
@@ -0,0 +1,40 @@
+package httpmw
+
+import (
+	"errors"
+
+	"github.com/casbin/casbin/v2"
+	"github.com/gin-gonic/gin"
+)
+
+var ErrNoUser = errors.New("missing user in request context")
+
+// CasbinRBACAuthorizeFunc uses a provided Casbin enforcer to implement RBAC middleware.
+// This function will look for roles stored in the gin.Context using the `groups` key
+// and will attempt to authorize the request using each found role.
+// If no match is made, the request will be rejected.
+func CasbinRBACAuthorizeFunc(e *casbin.Enforcer) RBACAuthorizeFunc {
```

Comment:
I would suggest changing the name a little bit, maybe `CasbinRBACAuthorizer` ? Although it makes sense to include `Func` suffix in the function type, I don't think there's much benefit to do so in the implementation.